R/chemspider.R

```
@@ -656,33 +376,79 @@ cs_convert <- function(query, from, to, apikey = NULL) {
     else {
       out <- out[, 1]
     }
+    return(out)
   }
-  else {
-    out <- unname(sapply(query, function(x) {
-      if (is.na(x)) return(NA)
-      switch(cs_convert_router(from, to),
-             identity = query,
-             cs_convert_multiple = cs_convert_multiple(
-               x, from = from, to = to, apikey = apikey
-             ),
-             cs_inchikey_csid = cs_inchikey_csid(x, apikey = apikey),
-             cs_inchi_csid = cs_inchi_csid(x, apikey = apikey),
-             cs_smiles_csid = cs_smiles_csid(x, apikey = apikey),
-             cs_mol_csid = stop("Conversion not supported."))
-      }))
+  if (to == "csid") {
+    if (from == "mol") {
+      if (verbose) message("Conversion not supported. Returning NA.")
+      return(NA_integer_)
     }
+    else {
+      out <- get_csid(query, from = from, apikey = apikey)$csid
+      return(out)
+    }
+  }
+  if (from == "inchikey" & to == "smiles") {
+    if (verbose) message("Conversion not supported. Returning NA.")
```

Just curious why you decided to make these messages and not errors. I would expect a combination of arguments that is not supported to result in an error or warning in most cases.

---

We had a discussion earlier about how functions should behave when the webservice is down. We concluded that in such a situation the function should fail fast. On second thought, you're right, this is a very similar situation and the function should return an error. Thanks for pointing this out.